libs/server/repository/core/application-services/src/lib/queries/get-single-repository-data/get-single-repository-data.handler.ts

```diff
@@ -19,6 +19,6 @@ export class GetSingleRepositoryDataHandler
       getRepoNameFromFullRepoName(query.fullName),
       getOwnerFromFullRepoName(query.fullName)
     );
-    return singleRepositoryDataReadModelFactory(repository);
+    return repository ? singleRepositoryDataReadModelFactory(repository) : null;
```

Comment:
the return type of this function is Promise<SingleRepositoryDataReadModel>. You cant return null from here.
If the repository doesn't exists throw an error.